게시글 상세 페이지 작성

rewind122 opened the pull request:
## PR 유형

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [x] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## PR 체크리스트

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

### PR 상세

게시글 상세 페이지 기능 구현 중입니다.
- 게시글 정보(사진, 텍스트, 작성자 프로필) 표시
- 댓글 정보(댓글, 댓글 작성자, 작성 시간) 표시
- 댓글 등록 기능: 스로틀링 혹은 디바운스 추후 적용 예정
- 스크롤 할 때마다 리렌더링이 되는 것 같은데 확인이 필요해보입니다... 

댓글 및 게시글 수정/삭제 기능은 아직 넣지 못했습니다ㅠㅠ
좀더 보완해서 머지할 수 있도록 더 수정할 예정입니다.

## 이슈

resolves #70

rewind122 committed:
refactor: 마이페이지 데이터 수정

- 현재 로그인 중인 유저의 데이터를 받아오도록 수정
- useGlobalStore의 initializeUser 수정

rewind122 committed:
refactor: 게시글 상세 페이지

- 게시글 데이터 불러오기
- 추가 버튼 클릭 이벤트/스타일링 수정
- 더보기 버튼 스타일링 수정

rewind122 committed:
refactor: 마이페이지 로딩 스피너 추가

rewind122 committed:
Merge branch 'develop' into feature/postDetail2/kgh

rewind122 committed:
refactor: 게시글 상세 페이지 기능 구현

- index 파일 분리
- postText 컴포넌트 분리 후 더보기 기능 추가
- 댓글 불러오기
- 댓글 시간 표시 함수 추가

hyesu-jeong reviewed (approved):
확인했습니다

castlehuni commented on src/pages/MyPage/index.jsx:
```diff
@@ -45,18 +50,18 @@ function MyPage() {
 
   const handleNewPost = () => {
-    location.href = '/게시글작성'; // 게시글 작성 페이지로 이동
+    location.href = '/writepost'; // 게시글 작성 페이지로 이동
```
location 보다는 useNavigate써주세요!!!

rewind122 replied:
넵!! 수정했습니다!

rewind122 committed:
refactor: 마이페이지 코드 수정

- location 코드를 useNavigate로 수정

rewind122 committed:
refactor: 게시글 상세 페이지 댓글 기능 추가

- 컴포넌트 분리 및 정리
- 댓글 추가 및 실시간 렌더링: input value가 초기화되지 않는 상태라 추후 다시 수정해야함
- 게시글 및 댓글 수정/삭제 기능 또한 추후 추가해야 함

rewind122 committed:
Merge branch 'develop' into feature/postDetail2/kgh
